enterprise/core-edge/src/main/java/org/neo4j/coreedge/core/state/InFlightLogEntryReader.java

```diff
@@ -59,10 +59,12 @@ public RaftLogEntry get( long logIndex ) throws IOException
             useInFlightMap = false;
             entry = getUsingCursor( logIndex );
         }
-
-        if ( removeReadIndexFromMap )
+        else
```

Comment:
I originally had an assertion inside the `apply` method which would only let you remove the lowest key and it would blow up if you tried to remove a key that wasn't in the map